streams/src/test/java/org/apache/kafka/streams/processor/internals/DefaultStateUpdaterTest.java

```diff
@@ -1043,6 +1046,28 @@ public void shouldResumeActiveStatefulTask() throws Exception {
         verify(changelogReader, times(2)).enforceRestoreActive();
     }
 
+    @Test
+    public void shouldAwaitWhenAllTasksPaused() throws Exception {
```

Comment:
This tests the absence of "restore" calls when all tasks are paused. This requires "smelly" constructs like `Mockito.reset` and `Thread.sleep`. We could alternatively not test this - the absence of something in a certain timeframe is not very amenable to unit testing.